src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
+const DAMAGE_FROM_CARNIVORE = 50;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
[CODE PERFORMANCE] - The `die` method currently uses the `indexOf` and `splice` methods to remove dead animals from the `Animal.alive` array. This approach can be improved by using the `filter` method to avoid nesting loops and potentially improve performance. Consider refactoring this method to use `filter` instead.